### PR TITLE
Client-side/server-side filters

### PR DESCRIPTION
**Pull request by venashial**
## Result:
image

----

## Toggle explanation:
image

---- 

Closes modrinth/knossos#100

### CODE REVIEW

File: pages/mods.vue [hunk 1]
```diff
@@ -504,11 +540,16 @@ export default {
         ]
 
         if (this.query.length > 0) {
+          console.log(this.query)
```
Comment (SwanX1) [inline pages/mods.vue:543]: does this debug line need to be here?

**Review by SwanX1 — COMMENTED** [review]
I don't believe the two debug `console.log`s should be there, but otherwise everything else looks good.

File: pages/mods.vue (hunk 1 above)
Comment (venashial) [reply pages/mods.vue]: Nope, my bad. Should be fixed with 62c3404b8cf8cd78b29f3b20a661ddf992cceb25

**Review by SwanX1 — APPROVED** [review]
Looks good to me, there are no lint errors when running `npm run lint`, and works on my local build as expected.

File: pages/mods.vue [hunk 2]
```diff
@@ -583,7 +652,6 @@ export default {
           window.history.replaceState(new Date(), 'Mods', url)
         }
       } catch (err) {
-        // eslint-disable-next-line no-console
```
Comment (Redblueflame) [inline pages/mods.vue:586]: Keep the eslint checker tho, as it could cause issues when having eslint enabled (for example in the CI runs).

**Review by Redblueflame — APPROVED** [review]
Thanks for this PR!
There is a small nitpick that could block the CI, but other than that, I'll be happy to merge it :)

File: pages/mods.vue (hunk 2 above)
Comment (venashial) [reply pages/mods.vue]: Sorry, I thought that I added those when I added my console.log()s. I added them back in 3d9be6d.

**Review by Mysterious-Dev — APPROVED** [review]

File: pages/mods.vue [hunk 3]
```diff
@@ -521,6 +561,34 @@ export default {
             formattedFacets.push(versionFacets)
           }
 
+          if (this.selectedEnvironments.length > 0) {
+            let environmentFacets = []
+            if (
```
Comment (falseresync) [inline pages/mods.vue:566]: This whole thing can be simplified:
```js
let includesClient = this.selectedEnvironments.includes('client');
let includesServer = this.selectedEnvironments.includes('server');
if (includesClient && includesServer) {
    environmentFacets = [
        ['client_side:required'],
        ['server_side:required'],
    ]
} else {
   ...
}
```
This way it is more visually appealing and makes less searches

**Review by falseresync — CHANGES_REQUESTED** [review]

File: pages/mods.vue (hunk 3 above)
Comment (venashial) [reply pages/mods.vue:566]: Will do, thanks.

**Review by falseresync — APPROVED** [review]

**Review by SwanX1 — APPROVED** [review]

**Comment by Geometrically** [comment]
Thanks for the PR :)

**Pull request merged**